src/irmin-pack/unix/mapping_file.ml

```diff
@@ -412,11 +412,12 @@ module Make (Io : Io.S) = struct
 
     (* Fill and close [file2] *)
     let poff = ref 0 in
-    let encode =
+    let encode i =
       let buf = Bytes.create 8 in
-      fun i ->
-        Bytes.set_int64_le buf 0 (Int64.of_int i);
-        Bytes.unsafe_to_string buf
+      Bytes.set_int64_le buf 0 (Int64.of_int i);
+      (* Bytes.unsafe_to_string is safe since [buf] will not be modified after
+         this function returns. We give up ownership. *)
+      Bytes.unsafe_to_string buf
```

Comment:
good catch, it should at least be documented. As usual, I'm worried about the perf of doing a lot of allocations, but maybe its fine.

Reply:
IIUC, to write this code correctly (based on OCaml's "ownership" discipline), I think we have to accept additional allocations. Looking at this particular code, we could allocate one buffer and append it instead of allocating three buffers per entry.

Reply:
that sounds like a good idea. Anyway, I don't want to block this PR, just to add a note here, if we ever see any issues in the benchmarks.